nasl/nasl_snmp.c

```diff
@@ -599,12 +721,46 @@ nasl_snmpv3_get (lex_ctxt *lexic)
       else if (!strcasecmp (privproto, "aes"))
         pproto = 1;
       else
-        return array_from_snmp_result (-2, "privproto should be des or aes");
+        {
+          nasl_perror (lexic, "Unsupported SNMP function.");
```

Comment:
```suggestion
          nasl_perror (lexic, "privproto should be des or aes");
```

Reply:
I messed up the position of this error message. This should be located within `nasl_snmpv1v2c_get`